scripts/prepare_nerfstudio.py

```diff
@@ -0,0 +1,160 @@
+"""Creates transforms.json file and resized images for running Nerfstudio from gtsfm output.
+
+This contains code from Nerfstudio for parsing data into the Nerfstudio format.
+https://github.com/nerfstudio-project/nerfstudio/blob/a121d76fd085f0fb356abf150c089c42eecbd066/nerfstudio/process_data/colmap_converter_to_nerfstudio_dataset.py#L29
+https://github.com/nerfstudio-project/nerfstudio/blob/a121d76fd085f0fb356abf150c089c42eecbd066/nerfstudio/process_data/colmap_utils.py#L16
+
+Author: Jon Womack
+"""
+import argparse
+import json
+import os
+from enum import Enum
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+from gtsam import Rot3
+
+from gtsfm.utils import images
+from gtsfm.utils import io as io_utils
+
+
+class CameraModel(Enum):
+    """Enum for camera types."""
+
+    OPENCV = "OPENCV"
+    OPENCV_FISHEYE = "OPENCV_FISHEYE"
+
+
+CAMERA_MODELS = {
+    "perspective": CameraModel.OPENCV,
+    "fisheye": CameraModel.OPENCV_FISHEYE,
+}
+
+
+def colmap_to_json(data_dir: str, output_dir: str, camera_model: CameraModel) -> Tuple[int, int]:
+    """Converts GTSfM's cameras.txt to to a JSON file, saves it in output_dir.
+    Args:
+        cameras_path: Path to the cameras.txt file.
+        images_path: Path to the images.txt file.
+        output_dir: Path to the output directory.
+        camera_model: Camera model used.
+    Returns:
+        Image dimension as a tuple.
+    """
+    wTi_list, input_images, calibrations, _, _, img_dims = io_utils.read_scene_data_from_colmap_format(data_dir)
+
+    frames = []
+    for i, image_name in enumerate(input_images):
+        c2w = wTi_list[i].matrix()
+
+        # Convert from GTSfM coordinate frame to nerfstudio coordinate frame.
+        # Invert camera X and Y, swap X and Y and invert Z in the world frame.
+        c2w[0:3, 1:3] *= -1
+        c2w = c2w[np.array([1, 0, 2, 3]), :]
+        c2w[2, :] *= -1
+
+        name = Path(f"./images/{image_name}")
+
+        frame = {
+            "file_path": name.as_posix(),
+            "transform_matrix": c2w.tolist(),
+        }
+        frames.append(frame)
+
+    # Only supports single intrinsics datasets.
+    camera_params = calibrations[0]
+    img_h, img_w = img_dims[0]
+
+    out = {
+        "fl_x": float(camera_params.fx()),
+        "fl_y": float(camera_params.fy()),
```

Comment:
would `fy()` exist for a radial camera calibration?

Reply:
yeah it is defined in the Cal3 base class, Cal3Bundler uses the same value for fx and fy